lib/kennel/models/project.rb

```diff
@@ -18,7 +18,10 @@ def self.file_location
       end
 
       def validated_parts
-        all = parts
+        # 'parts' called on same line as 'caller' for maximum cleanliness in strip_caller
+        here, all = [->() {caller}.call, parts]
+        all = [all] unless all.is_a?(Array)
```

Comment:
can do this or `all ||= []`
```suggestion
        all = Array(all)
```